files/__addonLocation__/package.json

```diff
@@ -36,6 +36,9 @@
     "@embroider/addon-shim": "^1.8.7",
     "decorator-transforms": "^2.0.0"
   },
+  "peerDependencies": {
+    "ember-source": ">= <%= latestEmberSource %>"
```

Comment:
What's the rationale to put the version hardcoded into index.js? We have versions everywhere right in the template file (here), so why deviate from this pattern? 

I would get it if the versions is determined in a smarter way than being hardcoded, but that's not the case. Seems more confusing to me to maintain tbh.

Reply:
I couldn't figure out how to get the version. Maybe i missed something? LatestVersion is async, and the locals object is sync

Reply:
IMO I would do same what we do in [ember-cli addon blueprint](https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon/files/addon-config/ember-try.js#L12) - make it second most recent LTS version

Reply:
Folks can change it if they want

Reply:
> Where? 😅

Like everywhere in https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/package.json? 😅 

We have all those versions specifiers hardcoded in that file, why not also ember-source?

> IMO I would do same what we do in [ember-cli addon blueprint](https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon/files/addon-config/ember-try.js#L12) - make it second most recent LTS version

I agree. Also it needs to match what we have in the [Readme](https://github.com/embroider-build/addon-blueprint/blob/main/files/README.md?plain=1#L7) and whatever ends up in `test-app/config/ember-try.js` as the minimum Ember version.

Reply:
Updated. Now at 5.4

Reply:
Do we want to drop 4.12? v1 blueprint still has it, fwiw...

If so, as mentioned above, we should also update [ember-try](https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon/files/addon-config/ember-try.js#L11-L18) and the generated Readme!

Reply:
Unrelated to this PR tho because we just need a template compiler.

Ember try still covers everything and this pr doesn't intend to change anything about that

Reply:
I do want to drop ts older than built in types tho

Reply:
Will happen in another pr to unblock built in types

Reply:
> Unrelated to this PR tho because we just need a template compiler.

I don't think that's true, as you are _introducing_ a peer dependency on ember-source here, but not on the version we supported until now, which is 4.12, but on >= 5.4. Which means, without violating the peer dependency, you are supposed to require Ember 5.4+ now. But then the addon setup would be very inconsistent, as ember-try and Readme still refer to 4.12.

Reply:
ah I see what you mean

Reply:
fixed https://github.com/embroider-build/addon-blueprint/pull/306/commits/58d2907f1f47e4d0a638e0e871515e73361e34bb